tools/generator/src/Generator/XCSchemeInfo+BuildActionInfo.swift

```diff
@@ -65,6 +89,27 @@ extension XCSchemeInfo.BuildActionInfo {
                     buildFor: buildTarget.buildFor
                 )
             }
-        try self.init(targets: buildTargetInfos)
+        let prePostActionsMapper = { (actions: [XcodeScheme.PrePostAction]) throws -> [XCSchemeInfo.PrePostAction] in
+            try actions.map { action in
+                let targetID = try targetIDsByLabel.value(
+                    for: action.target,
+                    context: "creating a `BuildActionInfo`"
+                )
+                return .init(name: action.name,
+                             target: try targetResolver.targetInfo(targetID: targetID),
+                             scriptContents: action.scriptContents)
+            }
+        }
```

Comment:
I suggest moving this to a [custom init like this one in BuildActionInfo](https://github.com/buildbuddy-io/rules_xcodeproj/blob/main/tools/generator/src/Generator/XCSchemeInfo%2BBuildActionInfo.swift#L49-L53).